sdk/core/azure-core/inc/azure/core/internal/http/pipeline.hpp

```diff
@@ -55,6 +57,100 @@ namespace Azure { namespace Core { namespace Internal { namespace Http {
       }
     }
 
+    /**
+     * @brief Construct a new Http Pipeline object from clientOptions.
+     *
+     * @remark The client options includes per retry and per call policies which are merged with the
+     * service-specific per retry policies.
+     *
+     * @param clientOptions The SDK client options.
+     * @param telemetryServiceName The name of the service for sending telemetry.
+     * @param telemetryServiceVersion The version of the service for sending telemetry.
+     * @param perRetryPolicies The service specific per retry policies.
+     */
+    explicit HttpPipeline(
+        ClientOptions const& clientOptions,
+        std::string const& telemetryServiceName,
+        std::string const& telemetryServiceVersion,
+        std::vector<std::unique_ptr<Azure::Core::Http::HttpPolicy>> const& perRetryPolicies)
+        : HttpPipeline(
+            clientOptions,
+            telemetryServiceName,
+            telemetryServiceVersion,
+            std::move(perRetryPolicies),
+            {})
+    {
+    }
+
+    /**
+     * @brief Construct a new Http Pipeline object from clientOptions.
+     *
+     * @remark The client options includes per retry and per call policies which are merged with the
+     * service-specific per retry policies.
+     *
+     * @param clientOptions The SDK client options.
+     * @param telemetryServiceName The name of the service for sending telemetry.
+     * @param telemetryServiceVersion The version of the service for sending telemetry.
+     * @param perRetryPolicies The service-specific per retry policies.
+     * @param perCallPolicies The service-specific per call policies.
+     */
+    explicit HttpPipeline(
+        ClientOptions const& clientOptions,
+        std::string const& telemetryServiceName,
+        std::string const& telemetryServiceVersion,
+        std::vector<std::unique_ptr<Azure::Core::Http::HttpPolicy>> const& perRetryPolicies,
+        std::vector<std::unique_ptr<Azure::Core::Http::HttpPolicy>> const& perCallPolicies)
+    {
+      auto const& perCallClientPolicies = clientOptions.GetPerCallPolicies();
+      auto const& perRetryClientPolicies = clientOptions.GerPerRetryPolicies();
+      // Adding 5 for:
+      // - TelemetryPolicy
+      // - RequestIdPolicy
+      // - RetryPolicy
+      // - LoggingPolicy
+      // - TransportPolicy
+      auto pipelineSize = perCallClientPolicies.size() + perRetryClientPolicies.size()
+          + perRetryPolicies.size() + perCallPolicies.size() + 3;
+
+      m_policies.reserve(pipelineSize);
+
+      // service-specific per call policies
+      for (auto&& policy : perCallPolicies)
```

Comment:
what does the `&&` after `auto` here mean?

Reply:
forward reference. `policy` will be a direct access to `perCallPolicies` instead, for instance, of creating a new reference `Policy&` to the lvalue.  
@CaseyCarter  did what I say make any sense?  🤣

Reply:
yeah, I get what you mean. But we all know `perCallPolicies` is const ref. So auto will be const lvalue reference. So
`for (auto& :)` or `for (const auto& :` will be enough. I just feel the extra `&` is kind of misleading.